scripts/deploy

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+#                                   #
+# Deploys the Hack as a Service API #
+#                                   #
+
+set -e
+
+# Error handler
+trap 'echo -e "\\n-------------------- SOMETHING WENT WRONG. --------------------\\n\\nYou may need to remove ~/.haas_deploy_lock before trying again."' ERR
+
+# Check/set deploy lock
+ssh -T -i $KEY_PATH deploy@$HAAS_IP << EOF
+    if [[ ! -f ~/.haas_deploy_lock ]];
+    then
+        # Set up stuff
+        rm -rf ~/haas_api
+        touch ~/.haas_deploy_lock
+    else
+        echo 'Deploy already in progress'
+        exit 1
+    fi
+EOF
+
+# Upload files to server
+echo "Uploading source to server..."
+rsync -zr --exclude ".git/" --exclude "target/" -e "ssh -i $KEY_PATH" . deploy@$HAAS_IP:~/haas_api
+
+ssh -T -i $KEY_PATH deploy@$HAAS_IP << EOF
+    set -e
+
+    cd ~/haas_api
+
+    # Build some images
+    docker build -t haas_api -f docker/prod.Dockerfile .
+    docker build -t haas_migrator -f docker/migrator.Dockerfile .
+
+    # Migrate the DB schema
+    echo -e "\nMigrating database..."
+    docker run --rm -e "DATABASE_URL=$DATABASE_URL" --network haas_admin haas_migrator
+
+    echo -e "\nCreating haas_api_next..."
+    HAAS_API_NEXT=\$(docker run -d \
+        --restart always \
+        --env-file ~/.env \
+        --network haas_admin \
+        --name haas_api_next \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        haas_api)
+    
+    HAAS_API_NEXT_IP=\$(docker inspect \$HAAS_API_NEXT | jq -r .[0].NetworkSettings.Networks.haas_admin.IPAddress)
+
+    echo "New container IP: \$HAAS_API_NEXT_IP"
+    
+    # Wait 5 seconds for the API to start up
+    echo -e "\nWaiting impatiently for HaaS to start up..."
+    sleep 5
```

Comment:
Ngl I'm a little sus of this - ideally we'd block until we can confirm for sure the API is up, but that'd be a little more difficult in a bash script. How long does it take for the API to start up now?

Reply:
It starts up instantly; the 5-second wait is just a safeguard

Reply:
Thanks!